packages/core/admin/admin/src/content-manager/history/components/HistoryAction.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { useQueryParams } from '@strapi/helper-plugin';
+import { Clock } from '@strapi/icons';
+import { stringify } from 'qs';
+import { useIntl } from 'react-intl';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+import type { DocumentActionComponent } from '../../../core/apis/content-manager';
+
+const StyledClock = styled(Clock)`
+  path {
+    fill: currentColor;
+  }
+`;
+
+const HistoryAction: DocumentActionComponent = ({ model, document }) => {
+  const { formatMessage } = useIntl();
+  const [{ query }] = useQueryParams<{ plugins?: Record<string, unknown> }>();
+  const navigate = useNavigate();
+  const pluginsQueryParams = stringify({ plugins: query.plugins }, { encode: false });
+
+  // TODO also check license
+  if (!window.strapi.future.isEnabled('history')) {
+    return null;
+  }
+
+  return {
+    icon: <StyledClock />,
+    label: formatMessage({
+      id: 'content-manager.history.document-action',
+      defaultMessage: 'Content History',
+    }),
+    onClick: () => navigate({ pathname: 'history', search: pluginsQueryParams }),
+    disabled: !document || !document.id || !model.startsWith('api::'),
```

Comment:
these disabled conditions can get complicated might be worth writing a list for the conditions when it's disabled e.g. `we're creating an entry, create a locale (where there will be an object for a document but no id), we only support content built in the CTB` etc.

I wouldn't have understood the last one about 6 months ago the concept of `api::` is typically abstracted from the FE.